internal/controller/finalizers_delete.go

```diff
@@ -121,8 +122,9 @@ func notifyOwnedResourceDeletion[T clusterOwnedResourceWithStatus](
 
 		if obj.GetStatusMessage() != statusMessage {
 			obj.SetAsFailed(errors.New(statusMessage))
-			if err := cli.Status().Patch(ctx, obj, client.MergeFrom(origObj)); err != nil {
-				itemLogger.Error(err, "error while setting failed status for cluster deletion")
+			obj.SetStatusObservedGeneration(0)
+			if err := cli.Status().Update(ctx, obj); err != nil {
```

Comment:
Note for reviewers: an update is needed here because the observed generation value would be removed by the `omitempty` serialization otherwise.

Reply:
```suggestion
			obj.SetStatusObservedGeneration(0)
			// We need to use an update here because of the observed generation set to 0
			// that would be ignored with the patch method.
			if err := cli.Status().Update(ctx, obj); err != nil {
```